internal/gitmetadata.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package internal
+
+import (
+	"os"
+	"sync"
+)
+
+const (
+	// EnvGitMetadataEnabledFlag specifies the environment variable name for enable/disable
+	EnvGitMetadataEnabledFlag = "DD_TRACE_GIT_METADATA_ENABLED"
+	// EnvGitRepositoryURL specifies the environment variable name for git repository URL
+	EnvGitRepositoryURL = "DD_GIT_REPOSITORY_URL"
+	// EnvGitCommitSha specifies the environment variable name git commit sha
+	EnvGitCommitSha = "DD_GIT_COMMIT_SHA"
+	// EnvDDTags specifies the environment variable name global tags
+	EnvDDTags = "DD_TAGS"
+
+	// TagRepositoryURL specifies the tag name for git repository URL
+	TagRepositoryURL = "git.repository_url"
+	// TagCommitSha specifies the tag name for git commit sha
+	TagCommitSha = "git.commit.sha"
+	// TagGoPath specifies the tag name for go module path
+	TagGoPath = "go_path"
+
+	// TraceTagRepositoryURL specifies the trace tag name for git repository URL
+	TraceTagRepositoryURL = "_dd.git.repository_url"
+	// TraceTagCommitSha specifies the trace tag name for git commit sha
+	TraceTagCommitSha = "_dd.git.commit.sha"
+	// TraceTagGoPath specifies the trace tag name for go module path
+	TraceTagGoPath = "_dd.go_path"
+)
+
+var (
+	lock = sync.Mutex{}
+
+	gitMetadataTags map[string]string
+)
+
+func updateTags(tags map[string]string, key string, value string) {
+	if _, ok := tags[key]; !ok && value != "" {
+		tags[key] = value
+	}
+}
+
+func updateAllTags(tags map[string]string, newtags map[string]string) {
+	for k, v := range newtags {
+		updateTags(tags, k, v)
+	}
+}
```

Comment:
> It's also not clear to me why values from src are ignored if they are ""? Is there a real use case for this?

It is for skip empty tags adding, they can appear e.g. because "" is default value for os.Getenv